miden/tests/integration/exec_iters.rs

```diff
@@ -0,0 +1,103 @@
+    });
+    let test = build_test!(source, &init_stack.clone());
+    let traces = test.execute_iter();
+    let fmp = Felt::new(1073741824);
+    let mem = vec![(1_u64, slice_to_word(&[13, 14, 15, 16]))];
+    let mut vm_states = vec![
+        VmState {
+            clk: 0,
+            stack: slice_to_felts(&[16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1]),
+            fmp: fmp,
+            memory: Vec::new(),
+        },
+        VmState {
+            clk: 1,
+            stack: slice_to_felts(&[16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 1]),
+            fmp: fmp,
+            memory: Vec::new(),
+        },
+        VmState {
+            clk: 2,
+            stack: slice_to_felts(&[1, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 3,
+            stack: slice_to_felts(&[16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 4,
+            stack: slice_to_felts(&[15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 5,
+            stack: slice_to_felts(&[14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 6,
+            stack: slice_to_felts(&[13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 7,
+            stack: slice_to_felts(&[12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 8,
+            stack: slice_to_felts(&[17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 9,
+            stack: slice_to_felts(&[17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0]),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+    ];
+    // Verify that memory is being written into addresses 1 and then 2 with the right values.
+    for t in traces {
+        let state = t.as_ref().unwrap();
+        let expected = vm_states.remove(0);
+        assert_eq!(expected, *state);
+    }
+
+    assert!(vm_states.is_empty());
+}
+
+// HELPER FUNCTIONS
+// =================================================================
+fn slice_to_word(values: &[i32]) -> [Felt; 4] {
+    [
+        Felt::new(values[0] as u64),
+        Felt::new(values[1] as u64),
+        Felt::new(values[2] as u64),
+        Felt::new(values[3] as u64),
+    ]
+}
+
+fn slice_to_felts(values: &[u64]) -> Vec<Felt> {
+    values.iter().map(|&v| Felt::from(v as u64)).collect()
+}
```

Comment:
We can probably use `ToElements` trait instead of this from the `core::utils` module. Once this is imported, you should be able to say something like `[1, 2, 3].to_elements()`

Reply:
fixed